tunnel.go

```diff
@@ -100,41 +126,62 @@ func NewTunnel(account *nkn.Account, identifier, from, to string, tuna bool, con
 				return nil, err
 			}
 		}
+
 		listeners = append(listeners, m)
 		err = m.Listen(config.AcceptAddrs)
 		if err != nil {
 			return nil, err
 		}
-		from = m.Addr().String()
-	} else {
-		listener, err := net.Listen("tcp", from)
-		if err != nil {
-			return nil, err
-		}
-		listeners = append(listeners, listener)
-	}
 
-	log.Println("Listening at", from)
+		f := m.Addr().String()
+		log.Println("Listening at", f)
+
+		toNKN := !strings.Contains(to[0], ":")
+
+		t := &Tunnel{
+			from:         f,
+			to:           to[0],
+			fromNKN:      fromNKN,
+			toNKN:        toNKN,
+			config:       config,
+			dialer:       dialer,
+			listeners:    listeners,
+			multiClient:  m,
+			tsClient:     c,
+			udpConnCache: cache.New(udpConnExpired, udpConnExpired),
+		}
+		tunnels = append(tunnels, t)
 
-	udpConnExpired := cache.NoExpiration
-	if config.UDPIdleTime > 0 {
-		udpConnExpired = time.Duration(config.UDPIdleTime) * time.Second
-	}
+	} else {
+		for i, f := range from {
```

Comment:
I think you can put for loop outside if/else to have less redundant code

Reply:
Yes modified.